Keas.Mvc/ClientApp/components/Equipment/EquipmentContainer.tsx

```diff
@@ -38,63 +51,141 @@ export default class EquipmentContainer extends React.Component<{}, IState> {
+    const allEquipmentList = this.context.person ? null : this.state.equipment;
     return (
       <div className="card">
         <div className="card-body">
                 <h4 className="card-title">Equipment</h4>
-                <EquipmentList equipment={this.state.equipment} onRevoke={this._revokeEquipment} />
-          <AssignEquipment onCreate={this._createAndMaybeAssignEquipment} />
+                <EquipmentList equipment={this.state.equipment} onRevoke={this._revokeEquipment} onAdd={this._assignSelectedEquipment} showDetails={this._openDetailsModal} />
+                <AssignEquipment
+                    onCreate={this._createAndMaybeAssignEquipment}
+                    modal={this.state.assignModal}
+                    modalLoading={this.state.assignModalLoading}
+                    openModal={this._openAssignModal}
+                    closeModal={this._closeAssignModal}
+                    selectedEquipment={this.state.selectedEquipment}
+                    selectEquipment={this._selectEquipment}
+                    unassignedEquipment={this.state.unassignedEquipment}
+                />
+                <EquipmentDetails selectedEquipment={this.state.selectedEquipment} modal={this.state.detailsModal} closeModal={this._closeDetailsModal} />
         </div>
       </div>
     );
   }
-  public _createAndMaybeAssignEquipment = async (equipment: IEquipment, person: IPerson) => {
-    // call API to create a equipment, then assign it if there is a person to assign to
+  private _createAndMaybeAssignEquipment = async (person: IPerson) => {
+      // call API to create a equipment, then assign it if there is a person to assign to
+      var equipment = this.state.selectedEquipment;
+      //if we are creating a new equipment
+      if (equipment.id === 0) {
+          equipment.teamId = this.context.team.id;
+          equipment = await this.context.fetch("/equipment/create", {
+              body: JSON.stringify(equipment),
+              method: "POST"
+          });
+      }
 
-    // TODO: basically all fake, make it real
-    let newEquipment: IEquipment = await this.context.fetch("/equipment/create", {
-      body: JSON.stringify(equipment),
-      method: "POST"
-    });
 
     // if we know who to assign it to, do it now
     if (person) {
-      const assignUrl = `/equipment/assign?equipmentId=${newEquipment.id}&personId=${
+      const assignUrl = `/equipment/assign?equipmentId=${equipment.id}&personId=${
         person.id
       }`;
 
-      newEquipment = await this.context.fetch(assignUrl, {
+      equipment = await this.context.fetch(assignUrl, {
         method: "POST"
       });
-    }
+      }
+
+    let index = this.state.equipment.findIndex(x => x.id == equipment.id);
+    console.log("index " + index);
+    if (index != -1) {
+        console.log("changing");
+        //update already existing entry in equipment
+        let updateEquipment = this.state.equipment.slice();
```

Comment:
@laholstege interesting, is this your way of making a shallow copy?  I've never seen anything like this but i guess it's the same as `let updateEquipment = [...this.state.equipment]`.  Is that what is intended?